geostore/permissions.py

```diff
@@ -32,3 +32,36 @@ def has_object_permission(self, request, view, obj):
                 return True
             else:
                 return has_perm
+
+
+class FeaturePermission(BasePermission):
+    def has_permission(self, request, view):
+        if request.method in SAFE_METHODS:
+            return True
+        else:
+            return request.user.has_perm('geostore.can_manage_layers')
+
+    def has_object_permission(self, request, view, obj):
+        if request.user.is_superuser:
+            return True
+
+        has_perm = request.user.has_perm('geostore.can_manage_layers')
+
+        if obj.layer.authorized_groups.exists():
+            # Checking permissions if the layer has group authorizations
+            is_authorized = any(map(
+                lambda group: group in obj.layer.authorized_groups.all(),
```

Comment:
I wound extract this queryset in a variable. Else the more the user has groups, the more queries are made.

Reply:
fixed/improved